pykeen/pykeen
# ♻️ 💉 Update docs and automate doctests

This PR does the following:

- Make sure more modules are automagically generated
- Add doctests to tutorials
- Update license year
- Update acknowledgements and author list
- Update typing in dataset to make sure it's possible to run the pipeline() on just a testing/training dataset

This PR supersedes #241 from @sbonner0. I'm sorry we never got around to figuring out the issue with GitHub actions, so I will include your name in this PR so you are credited with pointing out the original issue and providing the solution.
